release-engineering/platforms.md

```diff
@@ -0,0 +1,65 @@
+# Supported Platforms
+
+The Kubernetes release process produces [artifacts](./artifacts.md) for
+different architectures and operating systems. We consider the combination of
+architecture (`GOARCH`) and operating system (`GOOS`) as "platforms". Target of
+this document is to outline different categories of platforms as well as guiding
+through their graduation criteria.
+
+## Tiers
+
+Support for different platforms are organized into three tiers, whereas each
+comes with a different set of guarantees.
+
+### Tier 1
+
+Tier 1 platforms are considered as "guaranteed to work". To achieve this, they
+have to fulfil the following criteria:
+
+- **All** official binary releases are provided for the platform. This includes
+  container images as well as deb and rpm packages. Building the artifacts is
+  integrated in the release process.
+- Continuous Integration is set up to run tests for the platform. Necessary
+  tests are defined by SIG Release and usually correspond to the
+  [`blocking`](https://testgrid.k8s.io/sig-release-master-blocking) and
+  [`informing`](https://testgrid.k8s.io/sig-release-master-informing) testgrid
+  dashboards.
+- Documentation about the usage of artifacts for the platform is available.
+
+### Tier 2
+
+Tier 2 platforms are considered as "guaranteed to build". To achieve this, they
+have to fulfil the following criteria:
+
+- **Any** official binary release is provided for the platform. Building the
+  artifacts is integrated in the release process.
+- Automated testing is not or only partially setup.
+
+It may be possible that single features are not available for a certain
+platform.
+
+### Tier 3
+
+Tier 3 platforms are those which the Kubernetes codebase has support for, but
```

Comment:
What do you think "has support for" would encompass here? As it currently stands I think it could include _any_ platform, regardless of if there has been any effort to build Kubernetes for it. If that is the case, then I'm wondering how useful having a third tier is vs just calling these platforms unsupported. I would like to build out some tooling to be able to do experimental builds for new platforms we want to introduce. Maybe that could be tier 3?

Reply:
I see the point. There are some platforms which do not even support golang, so those would be definitely "unsupported". You're right, every valid golang platform which is not in tier 1 or 2 would be tier 3. It might build with version X, but since there is no CI in place it might break with future commits.

Reply:
Sounds good 👍